### DISCUSSION TITLE
different cookie for different authentication flows possible?

### DISCUSSION THREAD

**Discussion by GandalfUFR** [keycloak/keycloak#27664] [Q&A]
I am trying to find a solution to the use case of having different "browser auth flows" - one flow with second factor mandatory and the other one doesn't require a 2nd factor at all. Both flows are checking if cookie exists as a first step. If an user logs in first with a client that doesn't require a 2nd factor the user is automatically logged in to other clients where a 2nd factor should be required because of the existing cookie. What is the best solution to allow cookies in all flows but still force a 2nd factor for specific clients?

**Comment by ddurham2** [comment]
I would likewise question, if it's indeed a different question, can the cookie name differ when authenticating with a different client.  Presently, two web apps using the same realm, but different _clients_, will both behave as if logged in, if the cookie exists from having logged in with either web app.
